src/bindings/python/src/pyopenvino/core/infer_request.cpp

```diff
@@ -647,20 +650,26 @@ void regclass_InferRequest(py::module m) {
             :rtype: List[openvino.runtime.ConstOutput]
         )");
 
-    cls.def_property_readonly("inputs",
-                              &InferRequestWrapper::get_input_tensors,
-                              R"(
-                                Gets all input tensors of this InferRequest.
+    cls.def_property_readonly(
```

Comment:
Why do we need this API? We don;t have such API in C++.
Moreover it is not a backward compatible change. Before these methods returned tensors

Reply:
We have this API for `CompiledModel`: https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/core/compiled_model.cpp#L178-L185
The idea is to align the two. As I described in our internal chat, this indeed isn't a backward compatible change, the alternative is using deprecations and doing it over the span of potentially years. That's why I requested approval from both you and @ilya-lavrenov, to make sure we're aligned on which option we choose.

Reply:
do we have users of this property in OpenVINO Python notebooks?

Reply:
I searched the notebooks repo - `.inputs` and `.outputs` are not used for `InferRequest` objects, I didn't find even a single occurrence. However, they are frequently used for `CompileModel`, which we want alignment to.

Reply:
So CompiledModel in C++ also has information about model inputs, but InferRequest doesn't have. So we should align API between C++ and python and shouldn't align API between CompiledModel and InferRequest

Reply:
This change aligns Python with C++ in `InferRequest` case.

Before the change:
Python: `InferRequestObject.inputs`
C++: `InferRequestObject.get_input_tensors()`

After the change:
Python: `InferRequestObject.inputs`
C++: `InferRequestObject.m_inputs`

Reply:
It is not true.
Public C++ API doesn't have `m_inputs`: https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/openvino/runtime/infer_request.hpp
`m_inputs` is defined in python wrapper and has a type:
```
    std::vector<ov::Output<const ov::Node>> m_inputs;
    std::vector<ov::Output<const ov::Node>> m_outputs;
```
So it is not aligned with public C++ API, please do not mix public C++ API and some python wrappers.

Reply:
You changed the behavior of these methods, before it always return tensors, now you return vectors, so I don't understand why we need to change behaviour right now instead of removing this strange API

Reply:
I applied the changes we discussed. @ilya-lavrenov @ilyachur